File: 프로그래머스/연습문제/Lv1/최대공약수와 최소공배수/최대공약수와 최소공배수.cpp
#include <string>
#include <vector>

using namespace std;

int gcd(int n, int m){
    int l;
    
    while(m != 0){
        l = n%m;
        n = m;
        m = l;
    }
    
    return n;
}

vector<int> solution(int n, int m) {
    int g;
    
    if(n>m) g=gcd(m, n);
    else g=gcd(n, m);
    
    return {g, n*m/g};
}

File: BOJ/#2609 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def gcd(a, b):
    while b > 0:
        a,b = b,a%b

    return a

def main():
    a,b = map(int, input().split())
    print(gcd(a, b), a*b//gcd(a, b), sep='\n')

main()
